docs/data/tree-view/tree-item-customization/tree-item-customization.md

```diff
@@ -69,3 +126,112 @@ const CustomTreeItem2Content = styled(TreeItem2Content)(({ theme }) => ({
 ```
 
 :::
+
+## Hooks
+
+### `useTreeItem2`
+
+The `useTreeItem2` allows you to manage and customize individual tree items. You can use it to get the properties needed for all the slots, the status of the item or to tap into the interactive API of the Tree View.
+
+#### Slots properties
+
+The `useTreeItem2` hook offers granular control over the item's layout by providing resolvers to get the appropriate props for each slot. This allows you to build an entirely cutom layout for the tree items.
+
+The demo below shouls you how to get the props needed for each slot, and how to pass them correctly.
+
+{{"demo": "useTreeItem2HookProperties.js"}}
+
+You can pass additional props to a slot or override existing ones by passing an object an argument to the slot's props resolver.
+
+```jsx
+<CustomTreeItemContent
+  {...getContentProps({
+    className: 'overridingClassName',
+    newProp: 'I am passing this to the content slot'
+  })}
+>
+```
+
+#### Item status
+
+The `useTreeItem2` hook also returns a `status` object that holds boolean values for the different statuses of the tree item.
+
+```jsx
+const {
+  status: {
+    expanded, // True if he item is expanded.
+    expandable, // True if the item has children.
+    focused, // True if the item is focused.
+    selected, // True if the item is selected.
+    disabled, // True if the item is disabled.
+    editable, // True if the item is editable.
+    editing, // True if the item is being edited.
+  },
+} = useTreeItem2(props);
+```
+
+You can use these statuses to apply custom styling to the item, or conditionally render sub-components.
+
+{{"demo": "useTreeItem2HookStatus.js"}}
+
+#### Imperative API
+
+The `publicAPI` provides a number of methods to programatically interact with the Tree View. You can use the `useTreeItem2` hook to access the `publicAPI` object from within the Tree Item.
+
+{{"demo": "useTreeItem2HookPublicAPI.js"}}
+
+You can read more about the public API methods on each feature page of the Tree View.
+
+### `useTreeItem2Utils`
+
+The `useTreeItem2Utils` hook provides a set of interaction methods that allow you to implement custom behaviors for the tree view. It also returns the status of the item.
+
+```jsx
+const { interactions, status } = useTreeItem2Utils({
+  itemId: props.itemId,
+  children: props.children,
+});
+```
+
+In order to override the default interactions of the tree item, you can set `event.defaultMuiPrevented` to `true` in the event handlers, and implement your own behaviors.
+
+#### Selection
+
+You can select an item of the Tree View by clicking its content slot. The demo below demonstrates how to handle selection when the user clicks on an icon.
+
+{{"demo": "HandleSelectionDemo.js"}}
```

Comment:
I am struggling to find a decent use-case/ idea for a demo for `handleSelection` and `handleCheckboxSelection` -> so any suggestions are welcome here.